precompile/allow_list.go

```diff
@@ -125,6 +149,10 @@ func setAllowListRole(stateDB StateDB, precompileAddr, address common.Address, r
 	// Generate the state key for [address]
 	addressKey := address.Hash()
 	// Assign [role] to the address
+	// This stores the [role] in the contract storage with address [precompileAddr]
+	// and [addressKey] hash. It means that any reusage of the [addressKey] for different value
+	// conflicts with the same slot [role] is stored.
+	// Precompile implementations must use a different key than [addressKey]
```

Comment:
Good point, since this is something that a user of the `AllowList` needs to be aware of, can we add a comment to the AllowList type above?

In retrospect, I guess we should have used an allow list prefix, but if other implementations take that approach then it won't matter.

Reply:
added to precompile template